diff --git a/src/ce/keydownhandlers/ve.ce.LinearEnterKeyDownHandler.js b/src/ce/keydownhandlers/ve.ce.LinearEnterKeyDownHandler.js
index 70f72b93f..5173e6fc4 100644
--- a/src/ce/keydownhandlers/ve.ce.LinearEnterKeyDownHandler.js
+++ b/src/ce/keydownhandlers/ve.ce.LinearEnterKeyDownHandler.js
@@ -42,6 +42,7 @@ ve.ce.LinearEnterKeyDownHandler.static.execute = function ( surface, e ) {
 		cursor = range.from,
 		documentModel = surface.model.getDocument(),
 		emptyParagraph = [ { type: 'paragraph' }, { type: '/paragraph' } ],
+		emptyBreakLine = [ { type: 'break' }, { type: '/break' } ],
 		advanceCursor = true,
 		outermostNode = null,
 		nodeModel = null,
@@ -110,7 +111,10 @@ ve.ce.LinearEnterKeyDownHandler.static.execute = function ( surface, e ) {
 			);
 			advanceCursor = false;
 		}
-	} else if ( e.shiftKey && nodeModel.hasSignificantWhitespace() ) {
+	} else if ( !documentModel.sourceMode && e.shiftKey && nodeModel.getType() === 'paragraph' ) {
+		// Insert break node
+		txInsert = ve.dm.TransactionBuilder.static.newFromInsertion( documentModel, range.from, emptyBreakLine );
+	} else if ( !documentModel.sourceMode && e.shiftKey && nodeModel.hasSignificantWhitespace() ) {
 		// Insert newline
 		txInsert = ve.dm.TransactionBuilder.static.newFromInsertion( documentModel, range.from, '\n' );
 	} else if ( !node.splitOnEnter() ) {
diff --git a/src/ve.utils.js b/src/ve.utils.js
index 518e315d5..5931b7c75 100644
--- a/src/ve.utils.js
+++ b/src/ve.utils.js
@@ -818,7 +818,7 @@ ve.convertDomElements = function ( value ) {
 };
 
 ve.visibleWhitespaceCharacters = {
-	'\n': '\u21b5', // ↵
+	'\n': ' ', // space
 	'\t': '\u279e' // ➞
 };
 
